File: codes/mathematica/p112.m
(* 
 * Solution to Project Euler problem 112
 * Copyright (c) Sankha Saha. All rights reserved.
 * 
 *  
 *   
 *)


BouncyQ[n_] := Block[{d = IntegerDigits[n]},
  Block[{s = Sort[d]}, d != s && d != Reverse[s]]]

b = 0;
For[i = 1, True, i++,
  b += Boole[BouncyQ[i]];
  If[b / i == 99 / 100,
    Break[]]]
i

File: codes/python/p112.py
# 
# Solution to Project Euler problem 112
# Copyright (c) Sankha Saha. All rights reserved.
# 
#  
#   
# 

import itertools


def compute():
	count = 0
	for i in itertools.count(1):
		s = str(i)
		t = "".join(sorted(s))
		if s != t and s[ : : -1] != t:
			count += 1  # i is bouncy
		if count * 100 == 99 * i:
			return str(i)


if __name__ == "__main__":
	print(compute())
